src/explore-education-statistics-common/src/modules/table-tool/utils/__tests__/createExpandedColumnHeaders.test.ts

```diff
@@ -0,0 +1,1062 @@
+import Header from '../../components/utils/Header';
+import createExpandedColumnHeaders from '../createExpandedColumnHeaders';
+// these tests cover the test cases for headers that used to be in MultiHeaderDataTable
+
+describe('createExpandedColumnHeaders', () => {
+  test('should return an empty array if no column headers are provided', () => {
+    expect(createExpandedColumnHeaders([])).toEqual([]);
+  });
+
+  test('should return a single row of column headers if no groups are provided', () => {
+    const columnHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3'),
+    ];
+
+    expect(createExpandedColumnHeaders(columnHeaders)).toEqual([
+      [
+        { id: '1', text: '1', span: 1, crossSpan: 1, isGroup: false },
+        { id: '2', text: '2', span: 1, crossSpan: 1, isGroup: false },
+        { id: '3', text: '3', span: 1, crossSpan: 1, isGroup: false },
+      ],
+    ]);
+  });
+
+  test('should return multiple rows of column headers if groups are provided', () => {
+    const columnHeaders: Header[] = [
+      new Header('1', '1'),
+      new Header('2', '2'),
+      new Header('3', '3').addChild(new Header('2.1', '2.1')),
+      new Header('4', '4').addChild(new Header('3.1', '3.1')),
+    ];
+
+    expect(createExpandedColumnHeaders(columnHeaders)).toEqual([
+      [
+        {
+          crossSpan: 1,
+          id: '1',
+          isGroup: false,
+          span: 1,
+          text: '1',
+        },
+        {
+          crossSpan: 1,
+          id: '2',
+          isGroup: false,
+          span: 1,
+          text: '2',
+        },
+        {
+          crossSpan: 1,
+          id: '3',
+          isGroup: true,
+          span: 1,
+          text: '3',
+        },
+        {
+          crossSpan: 1,
+          id: '4',
+          isGroup: true,
+          span: 1,
+          text: '4',
+        },
+      ],
+      [
+        {
+          crossSpan: 1,
+          id: '2.1',
+          isGroup: false,
+          span: 1,
+          text: '2.1',
+        },
+        {
+          crossSpan: 1,
+          id: '3.1',
+          isGroup: false,
+          span: 1,
+          text: '3.1',
+        },
+      ],
+    ]);
+  });
+
+  describe('returns the correct headers with column groups & subgroups', () => {
+    test('returns the correct headers with one `colgroup` header subgroup', () => {
+      const columnHeaders: Header[] = [
+        new Header('B', 'B').addChild(
+          new Header('A', 'A')
+            .addChild(new Header('C', 'C'))
+            .addChild(new Header('D', 'D')),
+        ),
+      ];
+
+      const expandedColumnHeaders = [
```

Comment:
Same again - would use an explicit type here to get compile time feedback.

Same applies for rest of test cases.